PolyEngine/Engine/Src/ECS/ComponentIterator.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "Defines.hpp"
+
+namespace Poly
+{
+	template<typename T>
+	class Dynarray;
+	class ComponentBase;
+	template<typename PrimaryComponent, typename... SecondaryComponents>
+	struct IteratorProxy;
+
+	class IEntityIteratorHelper : public BaseObject<>
+	{
+		public:
+			virtual bool operator==(const IEntityIteratorHelper&) const = 0;
+			virtual bool operator!=(const IEntityIteratorHelper&) const = 0;
+			virtual Entity* get() const = 0;
+			virtual void increment() = 0;
+			virtual bool isValid() const = 0;
+	};
+
+	template<typename PrimaryComponent, typename... SecondaryComponents>
+	class ComponentIterator : public BaseObject<>,
+							public std::iterator<std::forward_iterator_tag, std::tuple<typename std::add_pointer<PrimaryComponent>::type, typename std::add_pointer<SecondaryComponents>::type...>>	
+	{
+		public:
+			explicit ComponentIterator(std::unique_ptr<IEntityIteratorHelper> iter) : Iter(std::move(iter))
+			{
+				UpdateCache();
```

Comment:
Just set cache to invalid. It'll update when needed.

Reply:
done